### ISSUE TITLE
Exception while using concurrent.futures.ProcessPoolExecutor and dask.multiprocessing, if tblib available

### ISSUE THREAD

**Issue by nmaxwell** [dask/dask#6619]
the root cause of this problem is this [issue](https://github.com/ionelmc/python-tblib/issues/62)

Now tblib doesn't appear to be very actively maintained, so I thought I should raise the same issue here, since dask depends on it. 

I put together a minimal docker file to show the problem:  
[Dockerfile.txt](https://github.com/dask/dask/files/5213532/Dockerfile.txt) (you'll have to rename from 'Dockerfile.txt' to 'Dockerfile', github won't me let upload otherwise)

to build it: `docker build -t map_bug .`

to run it, without dask to show the **expected output:**   `docker run -it map_bug python3 test.py`

output:

```
exception:  bar
 --- done --- 
```

to run it, with task, to show **the error:**  `docker run -it map_bug python3 test.py dask`

output:

```
exception:  bar
 --- done --- 
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 102, in _python_exit
    thread_wakeup.wakeup()
  File "/usr/lib/python3.8/concurrent/futures/process.py", line 90, in wakeup
    self._writer.send_bytes(b"")
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 183, in send_bytes
    self._check_closed()
  File "/usr/lib/python3.8/multiprocessing/connection.py", line 136, in _check_closed
    raise OSError("handle is closed")
OSError: handle is closed
```

The problem (from dask point of view) is in dask.multiprocessing.py, line 102, 

```python
try:
    import tblib.pickling_support

    tblib.pickling_support.install()

    def _pack_traceback(tb):
        return tb


except ImportError:

    def _pack_traceback(tb):
        return "".join(traceback.format_tb(tb))

    def reraise(exc, tb):
        exc = remote_exception(exc, tb)
        raise exc
```

so if tblib is available, dask will import it, which then causes the problem with concurrent.futures.ProcessPoolExecutor.

The python code running inside the above docker is:

```python
import sys
from concurrent.futures import ProcessPoolExecutor
if len(sys.argv) > 1 and sys.argv[1] == "dask":                     
    import dask.multiprocessing 

def foo(x):
    if x > 2:
        raise RuntimeError("bar") 

if __name__ == "__main__":
    with ProcessPoolExecutor(max_workers=2) as executor:

        try:
            list(
                executor.map(foo, range(50))
            ) 
        except Exception as ex: 
            print("exception: ", ex) 

    print(" --- done --- ")

```

**Environment**:

- Dask version: 2.25.0
- Python version: 3.8
- Operating System: ubuntu 20.02
- Install method (conda, pip, source): pip

**Comment by nmaxwell** [comment]
(I uploaded the wrong Dockerfile earlier, now corrected)

It's not fixed, it reproduces on python 3.8.5 and 3.8.2. He was running on Fedora, so that could influence things. Will test further. I supplied him with a reproducing Dockerfile also. 

Yeah our fix was to install 'dask[array]', which didn't include tblib. We could also have simply uninstalled it as you said. It's not really a problem with Dask per se, just thought you guys should be aware.
